### ISSUE TITLE
RequestValidator.validate uses req.originalUrl 

### ISSUE THREAD

**Issue by jtinbergen** [cdimascio/express-openapi-validator#242]
This results in a lot of cache entries in the middlewareCache because all the parameter values are taken into consideration too.

/customer/1234/detail/1289
/customer/1235/detail/1288

These two calls with the same method and content type create two entries in the cache. Perhaps the openAPI url definition /customer/{something}/detail/{somthingelse} would be a better key?

**Comment by cdimascio** [comment]
@jtinbergen thanks. This should get fixed. I like your proposal.
When `openapi.request.validator.ts` builds up the middleware cache, it has access to the `req.opeanpi` object. This contains

```javascript
 {expressRoute: "/v1/pets/:id", openApiRoute: "/v1/pets/{id}", ...}
```

This seems like a good candidate to replace `originalUrl` [here](https://github.com/cdimascio/express-openapi-validator/blob/master/src/middlewares/openapi.request.validator.ts#L73)

i'll look at this soon. In the meantime, if you're up to it, feel free to experiment as well. PRs are certainly welcome.

**Comment by cdimascio** [comment]
@jtinbergen this should be fixed in `3.9.0`. let me know how it goes. thanks!
